### DISCUSSION TITLE
Too low pandas_analyze_sample setting confusing error message

### DISCUSSION THREAD

**Discussion by Matklem** [duckdb/duckdb#4709] [General]
When working with large datasets (and especially sparse) in pandas and attempting to copy to CSV, Parquet or duckdb storage, the default pandas_analyze_sample setting might need to be adjusted.  Currently having a too low value only results in an error as:
`duckdb.InvalidInputException: Invalid Input Error: Failed to cast value: Unimplemented type for cast (VARCHAR -> NULL)`

A setting for sparse datasets would be nice, maybe traversing the column until X amount of non-null values have been evaluated.

At the least I suggest hinting towards this setting when throwing an error.

**Comment by craustin** [comment]
I expected casting the `object`-typed pandas DataFrame columns to `string` or `string[python]` would avoid this error since duckdb wouldn't need to analyze those columns to determine their type - but it seems it still does and I get this error when a string column has NAs for the first 1000 rows (for the default `pandas_analyze_sample=1000` setting.)

doc link: https://duckdb.org/docs/guides/python/import_pandas.html#object-columns

**Reply by Tishj** [reply]
I believe this is because Pandas uses Numpy internally, and Numpy can only handle `string` as `object` and has no separate type definition for it.
